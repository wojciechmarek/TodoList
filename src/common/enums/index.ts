export * from './theme';
export * from './task-status';
