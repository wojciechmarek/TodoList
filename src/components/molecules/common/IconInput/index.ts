export * from './icon-input';
export * from './icon-input.interface';
