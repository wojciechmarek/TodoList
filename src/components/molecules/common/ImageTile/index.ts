export * from './image-tile';
export * from './image-tile.interface';
