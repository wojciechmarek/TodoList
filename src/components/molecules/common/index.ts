export * from './IconButton';
export * from './IconInput';
export * from './ImageTile';
