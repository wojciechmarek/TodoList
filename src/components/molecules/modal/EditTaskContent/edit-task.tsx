import { Controller } from 'react-hook-form';
import { ModalEditTaskContentProps } from './edit-task.interface';
import {
  EditTaskContainer,
  EditTaskNameInput,
  EditTaskDescriptionTextArea,
} from './edit-task.styled';

export const ModalEditTaskContent = (props: ModalEditTaskContentProps) => {
  const { control } = props;
  return (
    <EditTaskContainer>
      <Controller
        name="title"
        control={control}
        render={({ field }) => (
          <EditTaskNameInput
            value={field.value}
            placeholder="Task name"
            handleInputChange={(value) => field.onChange(value)}
          />
        )}
      />
      <Controller
        name="description"
        control={control}
        render={({ field }) => (
          <EditTaskDescriptionTextArea
            value={field.value}
            placeholder="Task description"
            handleTextAreaChange={(value) => field.onChange(value)}
          />
        )}
      />
    </EditTaskContainer>
  );
};
