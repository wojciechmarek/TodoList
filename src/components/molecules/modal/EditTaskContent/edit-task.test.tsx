import { configure, render, renderHook } from '@testing-library/react';
import { ModalEditTaskContentProps } from './edit-task.interface';
import { UseFormReturn, useForm } from 'react-hook-form';
import { Task, TaskStatus } from '../../../../common';
import { ModalEditTaskContent } from './edit-task';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import TestRenderer from 'react-test-renderer';

describe('EditTaskContent', () => {
  let modalEditTaskContentProps: ModalEditTaskContentProps;
  let useFormResult: UseFormReturn<Task, any, undefined>;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    const { result } = renderHook(() =>
      useForm<Task>({
        defaultValues: {
          title: 'Go to the shop',
          description: 'Buy a milk',
          status: TaskStatus.Done,
          id: 123,
        },
      })
    );

    modalEditTaskContentProps = {
      control: result.current.control,
    };

    useFormResult = result.current;
  });

  it('should render', () => {
    const { baseElement } = render(
      <ModalEditTaskContent {...modalEditTaskContentProps} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(
      <ModalEditTaskContent {...modalEditTaskContentProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when the edit task modal content is rendered', () => {
    it('should render 2 text boxes (input and textarea)', () => {
      const { getAllByRole } = render(
        <ModalEditTaskContent {...modalEditTaskContentProps} />
      );
      const textBoxes = getAllByRole('textbox');
      expect(textBoxes.length).toBe(2);
    });

    it('should have correct initial value for input', () => {
      const { getAllByRole } = render(
        <ModalEditTaskContent {...modalEditTaskContentProps} />
      );
      const textBoxes = getAllByRole('textbox');
      const input = textBoxes.at(0);
      expect(input).toHaveValue('Go to the shop');
    });

    it('should have correct initial value for textarea', () => {
      const { getAllByRole } = render(
        <ModalEditTaskContent {...modalEditTaskContentProps} />
      );
      const textBoxes = getAllByRole('textbox');
      const textarea = textBoxes.at(1);
      expect(textarea).toHaveValue('Buy a milk');
    });
  });

  describe('when the edit task modal content is edited', () => {
    it('should update a value for input field', async () => {
      const { getAllByRole } = render(
        <ModalEditTaskContent {...modalEditTaskContentProps} />
      );
      const textBoxes = getAllByRole('textbox');
      const input = textBoxes.at(0) ?? new HTMLElement();
      await userEvent.type(input, 'qwerty', { delay: 1 });
      expect(useFormResult.getValues().title).toBe('Go to the shopqwerty');
    });

    it('should update a value for textarea field', async () => {
      const { getAllByRole } = render(
        <ModalEditTaskContent {...modalEditTaskContentProps} />
      );
      const textBoxes = getAllByRole('textbox');
      const textarea = textBoxes.at(1) ?? new HTMLElement();
      await userEvent.type(textarea, 'qwerty', { delay: 1 });
      expect(useFormResult.getValues().description).toBe('Buy a milkqwerty');
    });
  });
});
