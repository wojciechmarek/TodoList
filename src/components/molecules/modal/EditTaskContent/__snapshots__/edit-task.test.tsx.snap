// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTaskContent should match the snapshot 1`] = `
.emotion-0 {
  width: calc(100% - 4em);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 2em;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
}

.emotion-1 {
  border: 1px solid var(--search-input-border-and-icon-bg-color);
  background-color: var(--search-input-bg-color);
  color: var(--input-font-color);
  font-weight: 800;
  padding-left: 1em;
  border-radius: var(--standard-border-radius);
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  margin-bottom: 1em;
  height: 3em;
  box-shadow: var(--container-box-shadow);
}

.emotion-1::-webkit-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-1::-moz-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-1:-ms-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-1::placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-1:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.emotion-2 {
  border: 1px solid var(--search-input-border-and-icon-bg-color);
  background-color: var(--search-input-bg-color);
  color: var(--text-font-color);
  font-weight: 800;
  padding-left: 1em;
  padding-top: 0.5em;
  border-radius: var(--standard-border-radius);
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  height: calc(100% - 4em);
  max-width: 100%;
  min-width: 100%;
  box-shadow: var(--container-box-shadow);
}

.emotion-2::-webkit-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--text-font-color);
  font-weight: 600;
  opacity: 0.4;
}

.emotion-2:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<div
  className="emotion-0"
>
  <input
    className="emotion-1"
    onChange={[Function]}
    placeholder="Task name"
    value=""
  />
  <textarea
    className="emotion-2"
    onChange={[Function]}
    placeholder="Task description"
    value="Buy a milk"
  />
</div>
`;
